mesheryctl/pkg/utils/relationship.go

```diff
@@ -0,0 +1,224 @@
+
+		if relationship.Version == "*" {
+			modelBasePath := fmt.Sprintf("../server/meshmodel/%s", relationship.Model)
+			dirs, err := os.ReadDir(modelBasePath)
+			if err != nil {
+				err = utils.ErrReadDir(err, modelBasePath)
+				Log.Error(err)
+				continue
+			}
+
+			for _, dir := range dirs {
+				if dir.IsDir() {
+					versions = append(versions, dir.Name())
+				}
+			}
+
+			if len(versions) == 0 {
+				continue
+			}
+		} else {
+			versions = []string{relationship.Version}
+		}
+
+		for _, version := range versions {
+			if relationship.Model == "" {
+				continue
+			}
+			var rel _rel.RelationshipDefinition
+			rel.SchemaVersion = v1alpha3.RelationshipSchemaVersion
+			rel.Kind = _rel.RelationshipDefinitionKind(utils.ReplaceSpacesAndConvertToLowercase(relationship.KIND))
+			rel.RelationshipType = utils.ReplaceSpacesAndConvertToLowercase(relationship.Type)
+			rel.SubType = utils.ReplaceSpacesAndConvertToLowercase(relationship.SubType)
+			rel.EvaluationQuery = &relationship.EvalPolicy
+
+			rel.Version = "v1.0.0"
+			rel.Model = model.ModelDefinition{
+				Name:  relationship.Model,
+				Model: model.Model{Version: version},
+			}
+			status := _rel.Enabled
+			rel.Status = &status
+			var styles _rel.RelationshipDefinitionMetadataStyles
+
+			if relationship.Styles != "" {
+				err := encoding.Unmarshal([]byte(relationship.Styles), &styles)
+				if err != nil {
+					Log.Error(err)
+					continue
+				}
+				if rel.Metadata == nil {
+					rel.Metadata = &_rel.Relationship_Metadata{}
+				}
+			}
+
+			rel.Metadata = &_rel.Relationship_Metadata{
+				Description: &relationship.Description,
+				Styles:      &styles,
+			}
+
+			var selectorSet _rel.SelectorSet
+			if relationship.Selector != "" {
+				err := encoding.Unmarshal([]byte(relationship.Selector), &selectorSet)
+				if err != nil {
+					Log.Error(err)
+					continue
+				}
+				rel.Selectors = &selectorSet
+			}
+
+			fullPath, err := ConstructRelationshipPath(relationship.Model, version, rel.Version, "../server/meshmodel", rel)
+			if err != nil {
+				Log.Error(err)
+				continue
+			}
+
+			err = utils.WriteJSONToFile(fullPath, rel)
+			if err != nil {
+				Log.Error(err)
+				continue
+			}
+
+			Log.Info(fmt.Sprintf("Relationship written to %s", fullPath))
+		}
+	}
+}
+
+func ConstructRelationshipPath(modelName, version, defVersion, outputLocation string, rel _rel.RelationshipDefinition) (string, error) {
+	// Construct the base path using model and version
+	basePath := fmt.Sprintf("%s/%s/%s/v1.0.0", outputLocation, modelName, version)
+	if _, err := os.Stat(basePath); os.IsNotExist(err) {
+		return "", err
+	}
+
+	var fromKind, toKind string
+
+	if rel.Selectors != nil && len(*rel.Selectors) > 0 {
+		firstSelector := (*rel.Selectors)[0]
+		if len(firstSelector.Allow.From) > 0 && len(firstSelector.Allow.To) > 0 {
+			if firstSelector.Allow.From[0].Kind != nil {
+				fromKind = *firstSelector.Allow.From[0].Kind
+			}
+			if firstSelector.Allow.To[0].Kind != nil {
```

Comment:
This means there's a defined contract that in the sheet the relaationship definition will always have single selector, but why this constraint?